tests/test_postfix.py

```diff
@@ -202,26 +177,23 @@ def test_postfix_status(auto_container):
     )
 
 
-@pytest.mark.parametrize(
-    "container_per_test", POSTFIX_WITH_LOGGING_ENABLED, indirect=True
-)
-def test_postfix_send_email_delivered(
-    container_per_test: ContainerData,
-) -> None:
+def test_ppostfix_send_email_delivered(auto_container):
```

Comment:
Duplicate p. 
Also, shouldn't this use the `auto_container_per_test` fixture as the container is mutated during the test?

```suggestion
def test_postfix_send_email_delivered(auto_container):
```

Reply:
> Also, shouldn't this use the auto_container_per_test fixture as the container is mutated during the test?
I didn't check (before you pointed out) about this other fixture. Updated it now.

Btw - the mutation here change is the postfix config, and creation of `postfix.log` file?

Reply:
> Btw - the mutation here is change in the postfix config, and creation of `postfix.log` file?

Yes and sending an email. All of these actions modify the running container, which might or might not get reused by other test runs if you use the `container` or `auto_container` fixture. That can result in false positives or false negatives or might only show up in CIs (where very few parallel jobs are run, there container re-use in a single process is much more likely).